platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreConsensusEventFileManager.java

```diff
@@ -427,8 +427,15 @@ private long getNextSequenceNumber() {
      */
     public void finishedWritingFile(@NonNull final PreConsensusEventMutableFile file) {
 
+        final long previousFileHighestGeneration;
```

Comment:
Generally I would recommend a unit test or 2 for this new logic, but since we're bumping into release timelines I'll let you decide whether or not you want to write them

Reply:
added a new test